crates/scroll/alloy/provider/Cargo.toml

```diff
@@ -24,13 +25,22 @@ alloy-transport-http = { workspace = true, features = ["jwt-auth"] }
 scroll-alloy-network.workspace = true
 scroll-alloy-rpc-types-engine = { workspace = true, features = ["serde"] }
 
+# reth
+reth-rpc-api = { workspace = true, features = ["client"] }
+
+# reth-scroll
+reth-scroll-engine-primitives.workspace = true
```

Comment:
are we sure this won't cause circular dep? I've always tried to keep the alloy crate independent from the reth crates.